Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/LinuxTests.cs

```diff
@@ -0,0 +1,302 @@
+        {
+            return new FourCC('L', 'J', 'O', 'Y');
+        }
+
+        public static readonly string descriptorString = "{\"interface\":\"Linux\",\"type\":\"Joystick\",\"product\":\"TestProduct\",\"manufacturer\":\"TestManufacturer\",\"serial\":\"030000006d04000015c2000010010000\",\"version\":\"272\",\"capabilities\":\"{\\\"controls\\\":[{\\\"offset\\\":0,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":1,\\\"bit\\\":0},{\\\"offset\\\":0,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":2,\\\"bit\\\":1},{\\\"offset\\\":0,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":3,\\\"bit\\\":2},{\\\"offset\\\":0,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":4,\\\"bit\\\":3},{\\\"offset\\\":0,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":5,\\\"bit\\\":4},{\\\"offset\\\":0,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":6,\\\"bit\\\":5},{\\\"offset\\\":0,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":7,\\\"bit\\\":6},{\\\"offset\\\":0,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":8,\\\"bit\\\":7},{\\\"offset\\\":1,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":9,\\\"bit\\\":0},{\\\"offset\\\":1,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":10,\\\"bit\\\":1},{\\\"offset\\\":1,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":11,\\\"bit\\\":2},{\\\"offset\\\":1,\\\"featureSize\\\":1,\\\"featureType\\\":3,\\\"usageHint\\\":12,\\\"bit\\\":3},{\\\"offset\\\":4,\\\"featureSize\\\":4,\\\"featureType\\\":1,\\\"usageHint\\\":1,\\\"min\\\":-32768,\\\"max\\\":32767},{\\\"offset\\\":8,\\\"featureSize\\\":4,\\\"featureType\\\":1,\\\"usageHint\\\":2,\\\"min\\\":-32768,\\\"max\\\":32767},{\\\"offset\\\":12,\\\"featureSize\\\":4,\\\"featureType\\\":1,\\\"usageHint\\\":6,\\\"min\\\":-32768,\\\"max\\\":32767},{\\\"offset\\\":16,\\\"featureSize\\\":4,\\\"featureType\\\":1,\\\"usageHint\\\":7,\\\"min\\\":-32768,\\\"max\\\":32767},{\\\"offset\\\":20,\\\"featureSize\\\":4,\\\"featureType\\\":4,\\\"usageHint\\\":12,\\\"min\\\":-1,\\\"max\\\":1},{\\\"offset\\\":24,\\\"featureSize\\\":4,\\\"featureType\\\":4,\\\"usageHint\\\":13,\\\"min\\\":-1,\\\"max\\\":1}]}\"}";
+    }
+
+    [Test]
+    public void Layout_LinuxLayoutTest()
+    {
+        runtime.ReportNewInputDevice(TestSDLJoystick.descriptorString);
+
+        InputSystem.Update();
+
+        var generatedLayout = InputSystem.TryLoadLayout("Linux::TestManufacturer::TestProduct");
+
+        Assert.That(generatedLayout, Is.Not.Null);
+        Assert.That(generatedLayout.controls, Has.Count.EqualTo(16));
+
+        var triggerControl = generatedLayout.controls[0];
+        Assert.That(triggerControl.name, Is.EqualTo(SDLSupport.GetButtonNameFromUsage(SDLButtonUsage.Trigger)));
+        Assert.That(triggerControl.offset, Is.EqualTo(0));
+        Assert.That(triggerControl.layout, Is.EqualTo(new InternedString("Button")));
+    }
```

Comment:
@dbuckley3d A test like this is really hard to debug. Right now, it's failing because the count is 32 and not 16. It's hard to test why as it's not very clear what the test is actually testing. Instead of having a very long descriptor string with repetitive data, a better setup would be to have very specific and minimal test data that makes it very obvious what functionality is being established.